Dalamud/Game/Gui/ContextMenu/MenuTargetDefault.cs

```diff
@@ -0,0 +1,67 @@
+using Dalamud.Game.ClientState.Objects;
+using Dalamud.Game.ClientState.Objects.Types;
+using Dalamud.Game.ClientState.Resolvers;
+using Dalamud.Game.Network.Structures.InfoProxy;
+
+using FFXIVClientStructs.FFXIV.Client.UI.Agent;
+
+using Lumina.Excel.GeneratedSheets;
+
+namespace Dalamud.Game.Gui.ContextMenu;
+
+/// <summary>
+/// Target information on a default context menu.
+/// </summary>
+public sealed unsafe class MenuTargetDefault : MenuTarget
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="MenuTargetDefault"/> class.
+    /// </summary>
+    /// <param name="context">The agent associated with the context menu.</param>
+    internal MenuTargetDefault(AgentContext* context)
+    {
+        this.Context = context;
+    }
+
+    /// <summary>
+    /// Gets the name of the target.
+    /// </summary>
+    public string TargetName => this.Context->TargetName.ToString();
+
+    /// <summary>
+    /// Gets the object id of the target.
+    /// </summary>
+    public ulong TargetObjectId => this.Context->TargetObjectId;
+
+    /// <summary>
+    /// Gets the target object.
+    /// </summary>
+    public GameObject? TargetObject => Service<ObjectTable>.Get().SearchById(this.TargetObjectId);
+
+    /// <summary>
+    /// Gets the content id of the target.
+    /// </summary>
+    public ulong TargetContentId => this.Context->TargetContentId;
+
+    /// <summary>
+    /// Gets the home world id of the target.
+    /// </summary>
+    public ExcelResolver<World> TargetHomeWorld => new((uint)this.Context->TargetHomeWorldId);
```

Comment:
Can `World` be null for certain object types? If so, can you make this nullable and handle the case of `TargetHomeWorld` being invalid (zero?) by returning a `null`?

Reply:
Hrm. I'm unsure which would provide a better API experience then, seeing as most people tend to assume that real data resolves properly. At the very least, please add a docstring noting that this can be invalid (and therefore _actually_ return null).

Mostly trying to prevent surprised devs more than anything.

Reply:
If it can return null, the nullability definition should be used which is just `?` at the end